src/DI/ContainerBuilder.php

```diff
@@ -416,6 +416,8 @@ private function resolveImplement(ServiceDefinition $def, $name)
 						throw new ServiceCreationException("Type hint for \${$param->getName()} in $interface::$methodName() doesn't match type hint in $class constructor.");
 					}
 					$def->getFactory()->arguments[$arg->getPosition()] = self::literal('$' . $arg->getName());
+				} elseif (!$def->getSetup()) {
```

Comment:
I am not sure, what to choose.
So, @dg, please choose one and I'll update commit.